### PR TITLE
Bugfix/APPS-766 classification children job not picking rm types

### PR DESCRIPTION
**Pull request by n-insaidoo**
In this PR changes are introduced in order to evaluate the permission on nodes when the System user is reading.

### CODE REVIEW

File: repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    public void setup()
+    {
+//        queryEngine =(DBQueryEngine) this.applicationContext.getBean("dbQueryEngineImpl");
+//        permissionService = Mockito.spy(permissionService);
+//        queryEngine.setPermissionService(permissionService);
+        
+//        AclDAO aclDAO = (AclDAO) applicationContext.getBean("aclDAO");
+//        
+//        ((PermissionServiceImpl)permissionService).setAclDAO(aclDAO);
+
```
Comment (bbossola) [inline repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java:98]: Please remove unused code

File: repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+//        
+//        ((PermissionServiceImpl)permissionService).setAclDAO(aclDAO);
+        
+        prefix = "doc_"+System.currentTimeMillis() + "_";
+        authenticationComponent.setSystemUserAsCurrentUser();
+        
+//        StoreRef storeRef = nodeService.createStore(StoreRef.PROTOCOL_WORKSPACE, "Test_" + System.currentTimeMillis());
+//        St
+//        rootNodeRef = nodeService.getRootNode(storeRef);
+
```
Comment (bbossola) [inline repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java:105]: Please remove unused code

File: repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        return nodeRef;
+        
+    }
+    
+    @Test
+    public void shouldSelectNodesWhenExecutingQueryAsSystemUser() 
+    {
+        String query = "select * from cmis:document as d where d.cmis:name like '%"+prefix+"%'";
+//        when(permissionService.getReaders(anyLong())).thenReturn(Collections.emptySet());
+
```
Comment (bbossola) [inline repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java:166]: Please remove unused code

**Review by bbossola — CHANGES_REQUESTED** [review]

File: repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java (hunk 3 above)
Comment (n-insaidoo) [reply repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java]: My bad this class was not supposed to be committed.

File: repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java (hunk 1 above)
Comment (n-insaidoo) [reply repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java]: See https://github.com/Alfresco/alfresco-community-repo/pull/272#discussion_r570168257

File: repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java (hunk 2 above)
Comment (n-insaidoo) [reply repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java]: See https://github.com/Alfresco/alfresco-community-repo/pull/272#discussion_r570168257

**Review by bbossola — APPROVED** [review]

File: repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/NodePermissionAssessor.java [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+            
+            String owner = getOwner(node);
+            if (EqualsHelper.nullSafeEquals(authority.getAuthority(), owner))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
```
Comment (SaraAspery) [inline repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/NodePermissionAssessor.java:117]: you could just return `EqualsHelper.nullSafeEquals(authority.getAuthority(), owner)`

**Review by SaraAspery — APPROVED** [review]
just one really minor comment. LGTM

File: repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/NodePermissionAssessor.java (hunk 4 above)
Comment (n-insaidoo) [reply repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/NodePermissionAssessor.java]: You're right, now the method is returning  `EqualsHelper.nullSafeEquals(authority.getAuthority(), owner)` for that specific code block.

**Pull request merged**